### PR TITLE
test: increase test coverage to 100% for Tag

### LINKED ISSUE #17503: Increase `Tag` unit test coverage

### PR DESCRIPTION
**Pull request by 2nikhiltom**
Closes https://github.com/carbon-design-system/carbon/issues/17503

Increase test coverage to 100% for Tag component 

#### Changelog

Added new test cases 

#### Testing / Reviewing

Test coverage should increase
Verify that test makes sense

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
looks good to me!

File: packages/react/src/components/Tag/Tag-test.js [hunk 1]
```diff
@@ -6,11 +6,12 @@
  */
 
 import { Add } from '@carbon/icons-react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, fireEvent } from '@testing-library/react';
```
Comment (kennylam) [inline packages/react/src/components/Tag/Tag-test.js]:
```suggestion
import { render, screen } from '@testing-library/react';
```

Doesn't seem like this is being used? If needed consider [userEvent](https://testing-library.com/docs/user-event/intro/) instead.

Comment (2nikhiltom) [reply packages/react/src/components/Tag/Tag-test.js]: `fireEvent` was not used anywhere ,Good catch ! Thanks !! 🙌

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**